packages/itwin/tree-widget/src/test/trees/stateless/common/UseHierarchyVisibility.test.ts

```diff
@@ -0,0 +1,164 @@
+  };
+
+  beforeEach(() => {
+    visibilityHandler.getVisibilityStatus.reset();
+    visibilityHandler.changeVisibility.reset();
+    visibilityHandler.dispose.reset();
+  });
+
+  it("checks visibility status only once", async () => {
+    const node = createPresentationHierarchyNode({ id: "node-1" });
+    const { result } = renderHook(useHierarchyVisibility, { initialProps });
+
+    expect(visibilityHandler.getVisibilityStatus).to.not.be.called;
+    visibilityHandler.getVisibilityStatus.resolves({ state: "visible" });
+
+    act(() => {
+      // expect initial state to be `off` and disabled
+      const state = result.current.getCheckboxState(node);
+      expect(state.state).to.be.eq("off");
+      expect(state.isDisabled).to.be.true;
+    });
+
+    await waitFor(() => {
+      // wait for visibility status to calculated
+      expect(visibilityHandler.getVisibilityStatus).to.be.called;
+      const state = result.current.getCheckboxState(node);
+      expect(state.state).to.be.eq("on");
+    });
+
+    expect(result.current.getCheckboxState(node).state).to.be.eq("on");
+    expect(visibilityHandler.getVisibilityStatus).to.be.calledOnce;
+  });
+
+  it("recalculates node visibility status after visibility changed", async () => {
+    const node = createPresentationHierarchyNode({ id: "node-1" });
+    const { result } = renderHook(useHierarchyVisibility, { initialProps });
+
+    expect(visibilityHandler.getVisibilityStatus).to.not.be.called;
+    visibilityHandler.getVisibilityStatus.resolves({ state: "visible" });
+
+    act(() => {
+      // expect initial state to be `off` and disabled
+      const state = result.current.getCheckboxState(node);
+      expect(state.state).to.be.eq("off");
+      expect(state.isDisabled).to.be.true;
+    });
+
+    await waitFor(() => {
+      // wait for visibility status to calculated
+      expect(visibilityHandler.getVisibilityStatus).to.be.called;
+      const state = result.current.getCheckboxState(node);
+      expect(state.state).to.be.eq("on");
+    });
+
+    expect(visibilityHandler.getVisibilityStatus).to.be.calledOnce;
+    act(() => {
+      onVisibilityChange.raiseEvent();
+    });
+
+    visibilityHandler.getVisibilityStatus.resetBehavior();
+    visibilityHandler.getVisibilityStatus.resolves({ state: "partial" });
+
+    expect(visibilityHandler.getVisibilityStatus).to.be.calledOnce;
+    act(() => {
+      // expect visibility state to be same
+      expect(visibilityHandler.getVisibilityStatus).to.be.called;
+      const state = result.current.getCheckboxState(node);
+      expect(state.state).to.be.eq("on");
+    });
+
+    await waitFor(() => {
+      // wait for visibility status to calculated
+      expect(visibilityHandler.getVisibilityStatus).to.be.calledTwice;
+      const state = result.current.getCheckboxState(node);
+      expect(state.state).to.be.eq("partial");
+    });
+
+    expect(visibilityHandler.getVisibilityStatus).to.be.calledTwice;
```

Comment:
shouldn't we call `expect(result.current.getCheckboxState(node).state).to.be.eq("partial")` before this, similar to what previous test does?